lib/lightning/kafka_triggers/pipeline_supervisor.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Lightning.KafkaTriggers.PipelineSupervisor do
+  @moduledoc """
+  Supervisor of the BroadwayKafka pipelines that are created for each enabled
+  Kafka trigger.
+  """
+  use Supervisor
+
+  def start_link(_opts) do
+    Supervisor.start_link(__MODULE__, [], name: :kafka_pipeline_supervisor)
+  end
+
+  @impl true
+  def init(_opts) do
+    Supervisor.init([], strategy: :one_for_one)
```

Comment:
Considering this empty list, for children only started dynamically the `DynamicSupervisor` has the benefit of handling multiple pipelines more efficiently or, if there will be always a few number of kafka triggers, at least it handles shutdown more quickly (concurrently). Also using the `DynamicSupervisor` from the beginning would prevent the need to measure when it becomes more efficient on start up time and resource consumption.

Reply:
Hi @rorymckinley, the child specs (for GenServer ids) are accepted also by the `DynamicSupervisor`s.